libcontainerd/remote_unix.go

```diff
@@ -414,6 +414,18 @@ func (r *remote) runContainerdDaemon() error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+
+	// unless strictly necessary, do not add anything in between here
+	// as the reaper goroutine below needs to kick in as soon as possible
+	// and any "return" from code paths added here will defeat the reaper
+	// process.
+
+	r.daemonWaitCh = make(chan struct{})
+	go func() {
```

Comment:
mmm that `cmd.Wait()` will lay around to reap processes afaict and we can't wait on it with a channel (we'll hang the whole process otherwise), @mlaventure correct me if I'm wrong. With this PR we're not making sure `cmd.Wait` starts at all, we're making sure it starts really before any error code path kicks in and reaper won't work.